cloudformation/custom_resource/cluster.yaml

```diff
@@ -242,6 +259,8 @@ Resources:
               properties = event["ResourceProperties"]
               cluster_name = properties.get("ClusterName")
 
+              boto3.resource('s3').Bucket("${PclusterCustomResourceBucket}").objects.filter(Prefix=f"{cluster_name}/").delete()
```

Comment:
What happens if an exception is thrown above?

Reply:
The S3 cleanup custom resource fail to be deleted. Therefore, the provider CloudFormation stack fail to be deleted.
That said, because CloudFormation stack deletes resources as many as possible, pcluster custom resource alone with other resources are deleted

Reply:
Do you see any corner case where this code throws exception?

Reply:
Above there's a try-catch block that it seems we should have this as part of a `finally` or something for consistency.

Reply:
I looked into the code, we don't have to use any `try` `catch` as long as the code goes through the custom resource helper. The `try` `catch` you mentioned processes validation messages from pcluster. Here we do not need any process of the error message.

With your reminder I did add some `try` `catch` for code before going into the custom resource helper.